app/services/cosmos/cosmos.ts

```diff
@@ -195,14 +305,46 @@ export class CosmosAPI {
   createRewardsWithdrawMessage(
     fromAddress: string,
     validatorAddresses: string[],
-  ) {
-    return this.api.MultiMessage(
-      fromAddress,
-      validatorAddresses.map(validatorAddress =>
-        this.api.MsgWithdrawDelegationReward(fromAddress, {
+  ): CosmosMessage {
+    const msgs = validatorAddresses.map(validatorAddress => {
+      const withdrawMsg: MsgWithdrawDelegatorRewardEncodeObject = {
+        typeUrl: '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward',
+        value: MsgWithdrawDelegatorReward.fromPartial({
+          delegatorAddress: fromAddress,
           validatorAddress,
-        }),
-      ),
-    ) as CosmosMessage
+        })
+      }
+      return withdrawMsg
+    })
+    return { signerAddress: fromAddress, msgs }
+  }
+
+  /**
+   * Simulate gas according to messages' type and number
+   */
+  simulateGas(message: CosmosMessage): number {
+    let gas = 0
+    message.msgs.map(msg => {
+      switch (msg.typeUrl) {
+        case '/cosmos.bank.v1beta1.MsgSend':
+          gas += this.defaultGasLimits.send
+          break;
+        case '/cosmos.staking.v1beta1.MsgDelegate':
+          gas += this.defaultGasLimits.delegate
+          break;
+        case '/cosmos.staking.v1beta1.MsgBeginRedelegate':
+          gas += this.defaultGasLimits.transfer
+          break;
+        case '/cosmos.staking.v1beta1.MsgUndelegate':
+          gas += this.defaultGasLimits.undelegate
+          break;
+        case '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward':
+          gas += this.defaultGasLimits.withdraw
+          break;
+        default:
+          break;
```

Comment:
Should we have a default minimal gas? like using `transfer` * 2

Reply:
OK